org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/preferences/SonarLintGlobalConfiguration.java

```diff
@@ -238,25 +239,37 @@ public static Collection<RuleConfig> readRulesConfig() {
     return deserializeRulesJson(json);
   }
 
+  private static class RuleConfigGson {
+
+    enum Level {
+      @SerializedName("on")
+      ON,
+      @SerializedName("off")
+      OFF
+    }
+
+    Level level = Level.ON;
+    Map<String, String> parameters;
+
+  }
+
   private static Collection<RuleConfig> deserializeRulesJson(String json) {
     if (StringUtils.isBlank(json)) {
       return Collections.emptyList();
     }
-    var value = Json.parse(json);
+    var mapType = new TypeToken<Map<String, RuleConfigGson>>() {
+    }.getType();
+    Map<String, RuleConfigGson> rulesByKey = new Gson().fromJson(json, mapType);
     var result = new ArrayList<RuleConfig>();
     try {
-      var rulesJson = value.asObject();
-      rulesJson.forEach(ruleJson -> {
-        var ruleObject = ruleJson.getValue().asObject();
-        var rule = new RuleConfig(ruleJson.getName(), !RULE_JSON_LEVEL_OFF.equals(ruleObject.getString(RULE_JSON_LEVEL, RULE_JSON_LEVEL_ON)));
-        var ruleParamsValue = ruleObject.get(RULE_JSON_PARAMS);
-        if (ruleParamsValue != null) {
-          var paramsJson = ruleParamsValue.asObject();
-          paramsJson.forEach(p -> rule.getParams().put(p.getName(), p.getValue().asString()));
+      rulesByKey.forEach((key, config) -> {
+        var rule = new RuleConfig(key, RuleConfigGson.Level.OFF != config.level);
+        if (config.parameters != null) {
```

Comment:
This null check could be removed if map was initialised.

Reply:
This would produce a different output (see my other comment)